In [1]:
import pandas as pd

In [2]:
international_debt = pd.read_csv("international_debt.csv")

In [3]:
%%sql
SELECT SUM(debt) AS total_debt
FROM international_debt;

total_debt
0  3.079734e+12

In [4]:
%%sql
SELECT country_name, SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC
LIMIT 1;

country_name    total_debt
0        China  2.857935e+11

In [5]:
%%sql
SELECT country_name, SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt ASC
LIMIT 1;

country_name  total_debt
0  Sao Tome and Principe  44798032.5

In [7]:
%%sql
SELECT indicator_name, SUM(debt) AS total_debt
FROM international_debt
GROUP BY indicator_name
ORDER BY total_debt DESC;

indicator_name    total_debt
0   "Principal repayments on external debt, long-t...  7.322037e+11
1   "Principal repayments on external debt, privat...  4.077344e+11
2   "Disbursements on external debt, long-term (DI...  2.647011e+11
3        "PPG, official creditors (DIS, current US$)"  2.389960e+11
4   "Interest payments on external debt, long-term...  2.038590e+11
5         "PPG, private creditors (AMT, current US$)"  1.767620e+11
6        "PPG, official creditors (AMT, current US$)"  1.477073e+11
7                 "PPG, bilateral (DIS, current US$)"  1.382147e+11
8              "PPG, multilateral (DIS, current US$)"  1.007812e+11
9   "Interest payments on external debt, private n...  9.641246e+10
10                "PPG, bilateral (AMT, current US$)"  8.693960e+10
11                    "PPG, bonds (AMT, current US$)"  7.470105e+10
12        "PPG, private creditors (INT, current US$)"  7.053454e+10
13         "PPG, commercial banks (AMT, current US$)"  6.172897e+10
14             "PPG, multilateral (AMT, current US$)"  6.076771e+10
15                    "PPG, bonds (INT, current US$)"  5.552660e+10
16  "PPG, other private creditors (AMT, current US$)"  4.033200e+10
17       "PPG, official creditors (INT, current US$)"  3.691199e+10
18                "PPG, bilateral (INT, current US$)"  2.001938e+10
19             "PPG, multilateral (INT, current US$)"  1.689261e+10
20        "PPG, private creditors (DIS, current US$)"  1.650013e+10
21         "PPG, commercial banks (DIS, current US$)"  1.495856e+10
22         "PPG, commercial banks (INT, current US$)"  1.315840e+10
23  "PPG, other private creditors (INT, current US$)"  1.849535e+09
24  "PPG, other private creditors (DIS, current US$)"  1.541568e+09

### Summary of Findings

1. **Highest Total Debt:**
   - Country: China
   - Total Debt: \$285.79 billion

2. **Lowest Total Debt:**
   - Country: Sao Tome and Principe
   - Total Debt: \$44.80 million

3. **Debt by Indicator:**
   - The largest debt category is "Principal repayments on external debt, long-term (DOD, current US$)" with a total of \$732.20 billion.
   - The smallest debt category is "PPG, other private creditors (DIS, current US$)" with a total of \$1.54 billion.

### Detailed Queries and Results Interpretation

Let's go through the detailed steps and SQL queries used to arrive at these findings.

#### 1. Highest Total Debt by Country
To find the country with the highest total debt:

```sql
SELECT country_name, SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC
LIMIT 1;
```
**Result:**
- **Country:** China
- **Total Debt:** \$285.79 billion

#### 2. Lowest Total Debt by Country
To find the country with the lowest total debt:

```sql
SELECT country_name, SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt ASC
LIMIT 1;
```
**Result:**
- **Country:** Sao Tome and Principe
- **Total Debt:** \$44.80 million

#### 3. Debt by Indicator
To understand the distribution of debt across different categories:

```sql
SELECT indicator_name, SUM(debt) AS total_debt
FROM international_debt
GROUP BY indicator_name
ORDER BY total_debt DESC;
```
**Result:**
- The top category is "Principal repayments on external debt, long-term (DOD, current US$)" with \$732.20 billion.
- The bottom category is "PPG, other private creditors (DIS, current US$)" with \$1.54 billion.
